docs/development/pigs/pig-001.rst

```diff
@@ -0,0 +1,120 @@
+main Gammapy repository, and submitted as pull requests.
+
+Most discussions concerning Gammapy will happen by talking to each other directly
+(calls or face-to-face), or online on the mailing list or Github.
+If you're not sure if you should write a PIG, please don't! Instead bring the
+topic up in discussions first with other Gammapy developers, or on the mailing list,
+to get some initial feedback. This will let you figure out if writing a PIG
+will be helpful or not to realise your proposal.
+
+For now, we don't require a formal structure, apart from adding the author,
+creation date and status points at the top, as done at the top of this PIG.
+Please make your proposal clearly and keep the initial proposal short.
+If more information is needed, people that will review it will ask for it.
+
+Writing a PIG doesn't mean you have to implement it. That said, we expect that
+most PIGs will propose something that requires developments where someone has
+the intention to implement it within the near future (say within the next year).
+But it's not required, e.g. if you have a great idea or vision for Gammapy
+that requires a lot of development, without the manpower to execute the idea,
+writing a PIG could be a nice way to share this idea in some detail, with the
+hope that in collaboration with others it can eventually be realised.
+
+PIG review
+==========
+
+PIG review happens on the pull request on Github.
+
+When a PIG is put up, an announcement with a link to the pull request should
+be sent both to the Gammapy mailing list and the Gammapy coordinator list.
+
+Anyone is welcome to review it and is encouraged to share their thoughts
+in the discussion!
+
+Please note that Github hides inline comments after they have been edited,
+so we suggest that you use inline comments for minor points like spelling mistakes only.
+Put your main feedback as normal comments in the "Conversation" tab,
+so that for someone reading the discussion later they will see your comment directly.
+
+The final decision on any PIG is made by the Gammapy coordination committee.
+We expect that in most cases, the people participating in the PIG review will
+reach a consensus and the coordination committee will simply follow the outcome
+of the public discussion. But in unusual cases where disagreement remains,
+the coordination committee will talk to the people involved in the discussion
+with the goal to reach consensus or compromise, and then make the final decision.
+
+PIG status
+==========
+
+PIGs can have a status of:
+
+* "draft" - in draft status, either in the writing or dicussion phase
+* "withdrawn" - withdrawn by the author
+* "accepted" - accepted by the coordination committee
+* "rejected" - rejected by the coordination committee
+
+Final remarks
+=============
+
+This PIG leaves some points open. Specifically we don't say whether
+PIGs are accepted and merged before or after the actual implementation is done,
```

Comment:
This is somehow weird, if the PIG would be accepted after the actual implementation is done...the correct order is that a PIGs gets accepted and then implemented, no? If it's rejected or withdrawn it won't be implemented.

My suggestion would be to already be a *little* bit clearer about the process. E.g. open a pull request with a PIG, the PR is kept open until a decision about the PIG is made. Once the decision is made the PIG is assigned a status and the PR gets merged for the record. Modifications are possible at any time (as long as the PIG is not fully implemented yet) opening a new PR, where the status of the PIG might be revised once this PR gets merged.

Reply:
> the correct order is that a PIGs gets accepted and then implemented, no? 

Not necessarily, I think it can go either way. E.g. I think it would be awesome if someone tries to use https://www.tensorflow.org/ or https://github.com/HIPS/autograd for extremely efficient modeling and would be willing to contribute that to Gammapy in some form. But I would probably tell them to go implement it first or at least write an advanced prototype, and the PIG would say open until they come back with that. It's the same for many PEPs, usually Guido will only accept them if there's a full implementation already that is complete and stable and ready to be accepted.

So for Gammapy I think we don't want to commit here to anything before we have experience with PIGs. That's why it's so vague.

@adonath - OK? Or if not, could you please make concrete suggestions for changes to the PIG that you would find helpful or clarifying?